google-cloud-logging/src/main/java/com/google/cloud/logging/spi/DefaultLoggingRpc.java

```diff
@@ -148,21 +145,25 @@ public DefaultLoggingRpc(LoggingOptions options) throws IOException {
     }
   }
 
-  private static <V> Future<V> translate(ListenableFuture<V> from, final boolean idempotent,
-      int... returnNullOn) {
-    final Set<Integer> returnNullOnSet = Sets.newHashSetWithExpectedSize(returnNullOn.length);
-    for (int value : returnNullOn) {
-      returnNullOnSet.add(value);
+  private static <V> Future<V> translate(
+      RpcFuture<V> from, final boolean idempotent, Code... returnNullOn) {
+    final Set<Code> returnNullOnSet;
+    if (returnNullOn.length > 0) {
+      returnNullOnSet = EnumSet.of(returnNullOn[0], returnNullOn);
+    } else {
+      returnNullOnSet = Collections.<Code>emptySet();
     }
-    return Futures.catching(from, ApiException.class, new Function<ApiException, V>() {
-      @Override
-      public V apply(ApiException exception) {
-        if (returnNullOnSet.contains(exception.getStatusCode().value())) {
-          return null;
-        }
-        throw new LoggingException(exception, idempotent);
-      }
-    });
+    return from.catching(
+        ApiException.class,
+        new RpcFuture.Function<ApiException, V>() {
```

Comment:
Regarding the question in GAX about the location of `Function`: The notion of Function is not inherently tied to futures; what if we want to use them outside of futures?